src/components/AddressSearch.js

```diff
@@ -107,6 +107,14 @@ const AddressSearch = (props) => {
                 label: props.label,
                 containerStyles: props.containerStyles,
                 errorText: props.errorText,
+                onChangeText: (text) => {
+                    // Ensure whether an address is selected already has address value initialized.
+                    if (!_.isEmpty(googlePlacesRef.current.getAddressText())) {
+                        if (_.isEmpty(text) || !_.isEqual(text, props.value)) {
```

Comment:
Sorry, I'm confused why we're clearing the value if `text` is not the same as `props.value`. Can you help me understand?

Also NAB because this is personal preference, but I'm not a fan of the nested if statements – I think something like this would be slightly clearer:

```javascript
const isTextValid = !_.isEmpty(text) && _.isEqual(text, props.value);
if (!_.isEmpty(googlePlacesRef.current.getAddressText()) && !isTextValid) {
    saveLocationDetails({});
}
```